dsp/0002-state-of-the-metaverse.md

```diff
@@ -0,0 +1,312 @@
+
+* `position`: The player's position
+* `LoadedScenes`: The set of scenes that have been loaded or started loading
+* `WorldMapper`: A service that maps LAND coordinates to the most recent and valid Scene Manifest received.
+
+The algorithm itself has three steps:
+
+#### (1) Fetch and reconcile scenes to load:
+
+* Let `Coordinates` be the set of coordinates of all parcels in the visible radius from `position`
+* Let `ScenesToLoad` be an empty set of `Scenes`
+* For every `C` in `Coordinates`
+  * Let `S`, a Scene Manifest, be `WorldMapper.scene(C)`
+  * Add `S` to `ScenesToLoad`
+
+#### (2) Select which scenes should be loaded:
+* For every `S` in `ScenesToLoad`
+  * If `S` ∉ `LoadedScenes`:
+    * Load scene `S`
+    * Add `S` to `LoadedScenes`
+
+#### (3) Load and unload scenes
+* For every `S` in `LoadedScenes`
+  * If `S` ∉ `ScenesToLoad`
+    * Unload scene `S`
+    * Remove `S` from `LoadedScenes`
+
+### Parcel Manifest
+
+Every scene requires a number of configuration parameters and input information so user agents can display the scene correctly and connect to other clients. These fields include the IPFS CIDs of other materials required (models, scripts, textures, sounds...) as well as policy of how should the clients render and behave (whether flying is allowed, what are safe teleporting areas where clients might appear, which communications server to use to communicate with others...).
+
+This is an example parcel manifest that showcases some of those features:
+
+    {
+      "assets": {
+        "scene.js": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b",
+        "material_wood": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b",
+        "material_iron": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b",
+        "door_model": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b",
+        "favicon_asset": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b",
+        "background_music": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b",
+        "snapshot": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b",
+        "lod50": "/ipfs/QmZ1d5v83XSM7xbKDVAdFgS1PZmCzxuF5UtYrjMtFt6d6b"
+      },
+    
+      "contact": {
+        "name": "John Doe",
+        "email": "john@doe.com"
+      },
+    
+      "main": "scene.js",
+    
+      "scene": {
+        "base": "2,2",
+        "parcels": [
+          "1,2", "1,3", "1,4",
+          "2,2", "2,3", "2,4",
+          "3,2", "3,3", "3,4"
+        ]
+      },
+    
+      "communications": {
+        "type": "commsv2",
+        "server": "https://comms.decentraland.org"
+      },
+
+      "spawnPoints": [
+        {
+          "position": {
+            "x": 1.3, "y": 1.4, "z": 18.34
+          },
+          "rotation": {
+            "x": 0, "y": 180, "z": 0
+          },
+          "default": true
+        },
+        {
+          "area": {
```

Comment:
what is area?

Reply:
An area where it's safe to spawn if there are no other users around (a detail to prevent clipping/overlapping users)